src/typescript/frontend/src/components/trade/OrderEntry/MarketOrderEntry.tsx

```diff
@@ -0,0 +1,74 @@
+import { ApiMarket } from "@/types/api";
+import { OrderEntryInput } from "@/components/trade/OrderEntry/OrderEntryInput";
+import { type Side } from "@/types/global";
+import { Button } from "@/components/Button";
+import { ConnectedButton } from "@/components/ConnectedButton";
+import { useState } from "react";
+import { OrderEntryInfo } from "./OrderEntryInfo";
+
+export const MarketOrderEntry: React.FC<{
+  marketData: ApiMarket;
+  side: Side;
+}> = ({ marketData, side }) => {
+  // TODO: Replace with real market price
+  const [price, setPrice] = useState<string>("12.42");
+  const [amount, setAmount] = useState<string>("");
+
+  return (
+    <>
+      <div className="mx-4">
+        <OrderEntryInput
+          value={price}
+          onChange={setPrice}
+          startAdornment="MARKET PRICE"
+          endAdornment={marketData.quote.symbol}
+          type="number"
+          placeholder="0.00"
+          disabled
+        />
```

Comment:
I don't think the market order form should have a market price component. Average fill price will depend on the size of the order and I don't think it's worth trying to calculate it on the fly depending on what the user enters in the size input.